### PR TITLE
refactor: bench client

### PR DESCRIPTION
**Pull request by Kikkon**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
slove #486 
* what changes does this pull request make?
refactored the internal implementation of the beanch client."
* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
no

### CODE REVIEW

**Review by Phoenix500526 — COMMENTED** [review]
Please modify the code and make it pass the Clippy ci process.

**Comment by themanforfree** [comment]
Looks like you're referencing the wrong issue number, please update it.

**Review by Phoenix500526 — COMMENTED** [review]
Please modify your code to make it pass the clippy ci process. You can run `cargo clippy --all-targets --all-features -- -D warnings` in your local code.

File: benchmark/src/bench_client.rs [hunk 1]
```diff
@@ -90,16 +103,88 @@ impl BenchClient {
...
+        cluster.start().await;
+        let use_curp_client = true;
+        let config = ClientOptions::default();
+        let mut client = BenchClient::new(cluster.addrs(), use_curp_client, config)
+            .await
+            .unwrap();
+        //check xline clinet put value exist
+        let request = PutRequest::new("put", "123");
+        let _put_response = client.put(request).await;
+        match client.kv_client {
```
Comment (Phoenix500526) [inline benchmark/src/bench_client.rs:144]: Actually, you can implement a `get` method in `BenchClient` for testing only.

**Review by Phoenix500526 — CHANGES_REQUESTED** [review]

File: benchmark/src/bench_client.rs (hunk 1 above)
Comment (Kikkon) [reply benchmark/src/bench_client.rs:144]: Sounds good, I'll ping you again later after I've made the modifications.

File: benchmark/src/bench_client.rs [hunk 2]
```diff
@@ -91,16 +104,101 @@ impl BenchClient {
...
+        }
+    }
+
+    /// Send `RangeRequest` by `XlineClient` or `EtcdClient`
+    ///
+    /// # Errors
+    ///
+    /// If `XlineClient` or `EtcdClient` failed to send request
+    #[inline]
+    #[allow(dead_code)]
```
Comment (Phoenix500526) [inline benchmark/src/bench_client.rs:128]: Please use `#[cfg(test)]` to replace `#[allow(dead_code]`

File: benchmark/src/bench_client.rs [hunk 3]
```diff
@@ -91,16 +104,101 @@ impl BenchClient {
...
+        let use_curp_client = true;
+        let config = ClientOptions::default();
+        let mut client = BenchClient::new(cluster.addrs(), use_curp_client, config)
+            .await
+            .unwrap();
+        //check xline client put value exist
+        let request = PutRequest::new("put", "123");
+        let _put_response = client.put(request).await;
+        let range_request = RangeRequest::new("put");
+        match client.get(range_request).await {
```
Comment (Phoenix500526) [inline benchmark/src/bench_client.rs:169]: You can directly unwrap the result if you do believe that it's not an error. 
```suggestion
        let response = client.get(range_request).await.unwrap();
        assert_eq!(response.kvs[0].value, b"123");
```

File: benchmark/src/bench_client.rs [hunk 4]
```diff
@@ -90,16 +103,90 @@ impl BenchClient {
...
+        }
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_new_etcd_client() {
+        let mut cluster = Cluster::new(3).await;
+        cluster.start().await;
+        let use_curp_client = false;
+        let config = ClientOptions::default();
+        let mut client = match BenchClient::new(cluster.addrs(), use_curp_client, config).await {
```
[the remark this answers is not in the document]
Comment (Phoenix500526) [reply benchmark/src/bench_client.rs:185]: Same as above. 
```suggestion
        let mut client = match BenchClient::new(cluster.addrs(), use_curp_client, config).await.unwrap();
```

File: benchmark/src/bench_client.rs [hunk 5]
```diff
@@ -91,16 +104,101 @@ impl BenchClient {
...
+            Ok(client) => client,
+            Err(err) => {
+                panic!("should not be error: {err}");
+            }
+        };
+
+        let request = PutRequest::new("put", "123");
+        let _put_response = client.put(request).await;
+        let range_request = RangeRequest::new("put");
+        match client.get(range_request).await {
```
Comment (Phoenix500526) [inline benchmark/src/bench_client.rs:195]: Same as above.

File: benchmark/src/bench_client.rs [hunk 6]
```diff
@@ -6,7 +6,7 @@ use thiserror::Error;
 use xline::server::Command;
 use xline_client::{
     error::XlineClientError as ClientError,
-    types::kv::{PutRequest, PutResponse},
+    types::kv::{PutRequest, PutResponse, RangeRequest, RangeResponse},
```
Comment (Phoenix500526) [inline benchmark/src/bench_client.rs:9]:
```suggestion
#[cfg(test)]
use xline_client::types::kv::{RangeRequest, RangeResponse},
```

**Review by Phoenix500526 — CHANGES_REQUESTED** [review]

**Comment by Phoenix500526** [comment]
During the CI process of this PR, the test case `test_range_redirect ` panicked. I think it's a bug and not introduced by this PR. Could you please help to create a related issue to report this bug to the Xline community?

**Review by Phoenix500526 — APPROVED** [review]
LGTM

**Review by iGxnon — APPROVED** [review]
LGTM

**Pull request merged**